File: backend/app/models/comparison.py
from datetime import datetime

from pydantic import BaseModel, Field


class ComparisonRequest(BaseModel):
    paper_ids: list[str] = Field(..., min_length=2, max_length=5)


class ComparisonPaperSummary(BaseModel):
    paper_id: str
    title: str
    authors: list[str]
    source: str


class ComparisonDimension(BaseModel):
    label: str
    # One value per paper, in the same order as `ComparisonResult.paper_ids`.
    values: list[str]


class ComparisonResult(BaseModel):
    paper_ids: list[str]
    papers: list[ComparisonPaperSummary]
    dimensions: list[ComparisonDimension]
    assistant_take: str
    generated_at: datetime

File: frontend/types/comparison.ts
export interface ComparisonRequest {
  paper_ids: string[];
}

export interface ComparisonPaperSummary {
  paper_id: string;
  title: string;
  authors: string[];
  source: "arxiv" | "pubmed" | "upload";
}

export interface ComparisonDimension {
  label: string;
  /** One value per paper, aligned with `ComparisonResult.paper_ids` order. */
  values: string[];
}

export interface ComparisonResult {
  paper_ids: string[];
  papers: ComparisonPaperSummary[];
  dimensions: ComparisonDimension[];
  assistant_take: string;
  generated_at: string;
}
